src/documentation/system_administrators/advanced/quantum_safe_storage/quantum_safe_storage_deploy.md

```diff
@@ -0,0 +1,594 @@
+
+```sh
+apt update && apt install -y wget jq openssl
+```
+
+### Set Your EDITOR
+
+In the sections below, we'll be creating a number of config files. Set your preferred editor now to facilitate copying and pasting the commands. For example:
+
+```sh
+EDITOR=nano
+```
+
+### Install Tfcmd
+
+Here we will fetch the `tfcmd` binary and install it locally. It is released as part of `tfgrid-sdk-go`. You can find the latest releases [here](https://github.com/threefoldtech/tfgrid-sdk-go/releases). Choose the correct download link for your platform.
+
+For example, to install version 15.10 on a x64 Linux machine, you would run the following:
+
+```
+wget -O tfgrid-sdk-go.tar.gz https://github.com/threefoldtech/tfgrid-sdk-go/releases/download/v0.15.10/tfgrid-sdk-go_Linux_x86_64.tar.gz
+tar -C /usr/local/bin -xf tfgrid-sdk-go.tar.gz tfcmd
+chmod +x /usr/local/bin/tfcmd
+rm tfgrid-sdk-go.tar.gz
+```
+
+To test that it worked, you can use the following command:
+
+```sh
+tfcmd version
+```
+
+### Create Stub Zstor Config
+
+Next we will create a stub of the zstor config file. This will contain all of the information that needs to be filled manually. The final sections with the back end info will be filled automatically using scripts.
+
+Open a file `zstor-default.toml` in a text editor and paste in the template below. For example:
+
+```sh
+$EDITOR zstor-default.toml
+```
+
+Initial contents of the file:
+
+```
+minimal_shards = 16
+expected_shards = 20
+redundant_groups = 0
+redundant_nodes = 0
+zdbfs_mountpoint = "/mnt/qsfs"
+socket = "/var/run/zstor.sock"
+prometheus_port = 9100
+zdb_data_dir_path = "/data/data/zdbfs-data"
+max_zdb_data_dir_size = 25600
+
+[encryption]
+algorithm = "AES"
+key = "Write your key here"
+
+[compression]
+algorithm = "snappy"
+
+[meta]
+type = "zdb"
+
+[meta.config]
+prefix = "qsfs-meta"
+
+[meta.config.encryption]
+algorithm = "AES"
+key = "write your key here"
+```
+
+Make sure to edit the file as needed. You can change the minimal and expected shards according to your own plan. Another important value is `max_zdb_data_dir_size`, which is how large the cache is allowed to grow before data blocks are removed. This value is given in MiB. Therefore, the example shown is 25GiB.
+
+It is also necessary to fill in the encryption keys with your own. You can use the same or different keys for data and metadata, at your own preference. The key must be 32 bytes in hex format. Here's an example of how to generate a key in this format:
+
+```sh
+openssl rand -hex 32
+```
+
+Once this is done, take the output and insert it into the file in the indicated locations.
+
+### Prepare for Zdb Deployment
+
+Now we will deploy the zdbs and write their details into the config file in the proper format. Before proceeding, we'll set a few shell variables that will be used in the deployment scripts:
+
+```sh
+CONFIG=zstor-default.toml
+PASSWORD=$(openssl rand -base64 18)
```

Comment:
Password can't be `$(openssl rand -base64 18)`, I tried a few times and it failed, cause the password is too long.

```
WorkloadDeployError: WorkloadDeployError: Failed to deploy zdb with name zdb5 on node 159 due to: failed to create zdb namespace: failed to set password namespace 162-132876-zdb5 in 0-db: 2461494e-***: Argument too long.
```

Reply:
I could not reproduce this problem when deploying with `tfcmd`. I even increased the password length to 30+ characters without issue.